java-checks/src/main/java/org/sonar/java/checks/LeastSpecificTypeCheck.java

```diff
@@ -219,13 +226,11 @@ private Type leastSpecificType() {
       return longestChain.get(0);
     }
 
-    private static boolean isOverriding(Symbol s1, Symbol s2) {
-      return s1.isMethodSymbol()
-        && s2.isMethodSymbol()
-        && s1.name().equals(s2.name())
-        && ConfusingOverloadCheck.isPotentialOverride((Symbol.MethodSymbol) s1, (Symbol.MethodSymbol) s2);
+    private static boolean isOverridingWithSameReturnType(Symbol.MethodSymbol m1, Symbol.MethodSymbol m2) {
+      return m1.name().equals(m2.name())
+        && m1.returnType() == m2.returnType()
```

Comment:
The actual fix is to check that the return type is also the same in order to ensure that the least specific type is provides the exactly same methods being used